packages/preview-middleware-client/src/adp/controllers/AddFragment.controller.ts

```diff
@@ -274,14 +277,27 @@ export default class AddFragment extends BaseDialog<AddFragmentModel> {
             targetAggregation: targetAggregation ?? 'content'
         };
 
-        const command = await this.commandExecutor.getCommand(
+        const command = await this.commandExecutor.getCommand<AddFragmentChangeContentType>(
             this.runtimeControl,
             'addXML',
             modifiedValue,
             designMetadata,
             flexSettings
         );
 
+        const templateName = this.getFragmentTemplateName(modifiedValue.targetAggregation);
+        if (templateName) {
+            const preparedChange = command.getPreparedChange();
+            const content = preparedChange.getContent();
+            preparedChange.setContent({ ...content, templateName });
```

Comment:
I would suggest making `getPreparedChange` (and possibly `getContent` too if needed) a generic arrow function. 

Example
```ts
getPreparedChange: <T>() => T;
```

This way you can pass the whole **AddXMLChange** interface which is defined in the types.ts in `adp-tooling` and have type safety.

Reply:
It might be that we need to add generics to the functions in any case as there could be slightly different changes created for the same command, but I think it makes more sense to deal with this holistically in a separate task and I don't think we'll get it right here just by making something for this one example.